File: .claude/skills/write-orchestration-script/tests/scenarios/shape_smoke.py
#!/usr/bin/env python3
"""Python shape compose-smoke: drive the dispatcher through one item against the
mock's `happy` scenario, and import scheduler/deliberation/pipeline to prove they
load. Emits TRACE {"event":"smoke","ok":true}. TEST harness only."""

import importlib.util
import json
import os
import sys

_ASSETS = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", "assets", "python"))
sys.path.insert(0, _ASSETS)

from lares_client import LaresClient  # noqa: E402
import dispatcher  # noqa: E402


def _import(name):
    spec = importlib.util.spec_from_file_location(name, os.path.join(_ASSETS, f"{name}.py"))
    mod = importlib.util.module_from_spec(spec)
    spec.loader.exec_module(mod)
    return mod


# import-load the other shapes (they must not auto-run under import)
for shape in ("scheduler", "deliberation", "pipeline"):
    _import(shape)

client = LaresClient.connectApi()
results = dispatcher.run_dispatcher(
    client, [{"id": "smoke-item"}],
    item_to_work=lambda item: {
        "payload": {"title": f"smoke {item['id']}", "provider": "claude", "isSupervised": True},
        "kickoff": "do smoke work; end with DONE",
        "artifact": None, "baseline_hash": None,
    },
    accept_token=lambda t: True,
    concurrency=1,
)
ok = len(results) == 1 and results[0]["agentId"] is not None
sys.stdout.write("TRACE " + json.dumps({"event": "smoke", "ok": ok, "results": results}) + "\n")
sys.exit(0 if ok else 1)

File: .claude/skills/write-orchestration-script/tests/scenarios/shape_smoke.mjs
// Node shape compose-smoke: drive the dispatcher through one item against the
// mock's `happy` scenario (client → launch → kickoff → complete → retire), and
// exercise the control-skeleton import. Emits TRACE {event:"smoke",ok:true}.
// TEST harness only.

import path from 'node:path';
import { fileURLToPath, pathToFileURL } from 'node:url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const base = path.resolve(__dirname, '..', '..', 'assets', 'node');
const url = (f) => pathToFileURL(path.join(base, f)).href;
const { LaresClient } = await import(url('lares-client.mjs'));
const dispatcher = await import(url('dispatcher.mjs'));
await import(url('control-skeleton.mjs')); // import-loads (no auto-run)

const client = await LaresClient.connectApi();
// token-only deliverable (no artifact) — the happy mock ends turns with a message.
const results = await dispatcher.runDispatcher(client, [{ id: 'smoke-item' }], {
  itemToWork: (item) => ({
    payload: { title: `smoke ${item.id}`, provider: 'claude', isSupervised: true },
    kickoff: 'do smoke work; end with DONE',
    artifact: null,
    baselineHash: null,
  }),
  acceptToken: () => true, // happy mock returns "done"; accept any completion token
  concurrency: 1,
});
const ok = results.length === 1 && results[0].agentId != null;
process.stdout.write('TRACE ' + JSON.stringify({ event: 'smoke', ok, results }) + '\n');
process.exit(ok ? 0 : 1);
